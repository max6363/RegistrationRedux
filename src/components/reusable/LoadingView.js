import React from 'react';
import {View, Text, ActivityIndicator} from 'react-native';

const LoadingView = props => {
  const {container, innerView, textStyle} = styles;
  return (
    <View style={container}>
      <View style={innerView}>
        <ActivityIndicator animating={true} size="large" color="black" />
        {props.text ? <Text style={textStyle}>{props.text}</Text> : null}
      </View>
    </View>
  );
};

const styles = {
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    backgroundColor: '#00000050',
  },
  innerView: {
    backgroundColor: 'white',
    padding: 10,
    borderRadius: 10,
  },
  textStyle: {
    marginTop: 20,
    color: '#00000090',
    fontSize: 30,
  },
};

export default LoadingView;
