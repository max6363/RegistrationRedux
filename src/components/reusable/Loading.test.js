import React from 'react';
import {shallow} from 'enzyme';
import LoadingView from './LoadingView';

describe('Test LoadingView', () => {
  it('can test LoadingView with text', () => {
    const wrapper = shallow(<LoadingView text="loading" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('can test LoadingView without text', () => {
    const wrapper = shallow(<LoadingView />);
    expect(wrapper).toMatchSnapshot();
  });
});
