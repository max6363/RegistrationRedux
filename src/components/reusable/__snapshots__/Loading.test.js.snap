// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoadingView can test LoadingView with text 1`] = `ShallowWrapper {}`;

exports[`Test LoadingView can test LoadingView without text 1`] = `ShallowWrapper {}`;
